src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php

```diff
@@ -101,6 +92,31 @@ protected function processValue($value, bool $isRoot = false)
         return new Reference($id);
     }
 
+    /**
+     * @param array<array-key, mixed> $services
+     *
+     * @return array<string, mixed>
+     */
+    private function processLocatorKeys(array $services): array
+    {
+        $i = 0;
+        foreach ($services as $k => $v) {
+            if ($i === $k && $v instanceof Reference) {
+                unset($services[$k]);
+                $k = (string) $v;
+                ++$i;
+            } elseif (\is_int($k)) {
+                $i = null;
+            }
+
+            $services[$k] = $v;
+        }
+
+        ksort($services);
```

Comment:
Why is this needed?

Reply:
If I knew... I extracted this code for reusing. The `ksort()` was removed from the `register` method on c67c2df2362af3a317dadfa54d6038cfdbf07230. I have no idea why it still exists in `processValue`.